library/core/src/intrinsics.rs

```diff
@@ -846,6 +846,10 @@ extern "rust-intrinsic" {
     /// destination value, then forgets the original. It's equivalent to C's
     /// `memcpy` under the hood, just like `transmute_copy`.
     ///
+    /// Because `transmute` is a by-value operation, alignment is not a concern.
+    /// As with any other function, the compiler already ensures both `T` and `U`
+    /// are properly aligned.
```

Comment:
This could be interpreted to mean that I can transmute `&u8` to `&u32` as "alignment is not a concern"...

So if we really have to talk about alignment, I think we have to be very clear. Maybe something like
```
    /// Because `transmute` is a by-value operation, alignment of the transmuted values is not a concern.
    /// As with any other function, the compiler already ensures both `T` and `U`
    /// are properly aligned. However, when transmuting pointers, the caller has to ensure
    /// proper alignment of the pointed-to values.
```

Reply:
Applied your suggestion along with notes that it's not just pointers, references and types like boxes as well. Thanks.